src/agent/coverage/src/cobertura.rs

```diff
@@ -32,30 +52,47 @@ pub fn cobertura(source_coverage: SourceCoverage) -> Result<String, Error> {
     )?;
 
     emitter.write(XmlEvent::start_element("packages"))?;
-    emitter.write(
-        XmlEvent::start_element("package")
-            .attr("name", "0")
-            .attr("line-rate", "0")
-            .attr("branch-rate", "0")
-            .attr("complexity", "0"),
-    )?;
-
-    emitter.write(XmlEvent::start_element("classes"))?;
-    // loop through files
-    let files: Vec<SourceFileCoverage> = source_coverage.files;
-    for file in files {
+    // loop through files, path (excluding file name) will be package name for better results with ReportGenerator
+    let package_files: Vec<SourceFileCoverage> = copy_source_coverage.files;
+    let mut package_valid_lines = 0;
+    let mut package_hit_lines = 0;
+    let mut package_line_rate = 0_f32;
+    for file in package_files {
+        let copy_file = file.clone();
+        let package_locations: Vec<SourceCoverageLocation> = file.locations;
+        for location in package_locations {
+            package_valid_lines += 1;
+            if &location.count > &0 {
+                package_hit_lines += 1;
+            }
+        }
+        if package_valid_lines > 0 {
+            package_line_rate = package_hit_lines as f32 / package_valid_lines as f32;
+        }
+        let full_path_file = Path::new(&file.file);
+        let path = full_path_file.parent().unwrap();
+        emitter.write(
```

Comment:
Ideally, the data aggregation (line/hit/rate computation) and markup codegen (XML element emitting) will be really clearly separated. If you factor out the aggregation that I mentioned, you could perhaps introduce a new struct return value that would make it easy to do this. The resulting code could be a lot easier to understand or change in the future!

Reply:
Split out into a couple different functions, wasn't able to completely unduplicate.